plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,101 @@
+/*##############################################################################
+1
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems�.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "nlp_eng.hpp"
+#include "nlp_engine.h"
+
+#ifdef _WIN32
+#include<direct.h>
+#else
+#include <unistd.h>
+#endif
+#include <filesystem>
+using namespace std;
+
+#define NLP_DEBUG
+#ifdef NLP_DEBUG
+#include <iostream>
+#define NLP_DEBUG_FILE "/tmp/nlp_debug.txt"
+#endif
+static NLP_ENGINE *nlpEngine = NULL;
+
+NLPEng::NLPEng(const char *workingFolderPath) {
+
+#ifdef NLP_DEBUG
+    ofstream handle;
+    handle.open(NLP_DEBUG_FILE, std::ofstream::out | std::ofstream::app);
+#endif
+
+    if (workingFolderPath[0]) {
+        _stprintf(workingFolder,"%s/%s",workingFolderPath,"nlp-engine");
+
+    } else {
+        char working[FILENAME_MAX+1];
+        readlink("/proc/self/exe", working, FILENAME_MAX);
+
+#ifdef NLP_DEBUG        
+        handle << "[readlink: " << working << "]" << endl;
+#endif
+        string w = working;
+        size_t pos = w.find_last_of("/");
+        size_t pos2 = w.find_last_of("/",pos-1);
+        string parent = w.substr(0,pos2);
+        _stprintf(workingFolder,"%s/%s",parent.c_str(),"plugins/nlp/nlp-engine");
+    }
+
+#ifdef NLP_DEBUG    
+    handle << "[workingFolder: " << workingFolder << "]" << endl;
+    handle.close();
+#endif
+}
+
+NLPEng::~NLPEng() {
+    if (nlpEngine != NULL) {
+        delete nlpEngine;
+    }
```

Comment:
As per RKCs previous comment this can be changed to:
```
delete nlpEngine;
nlpEngine = NULL;
```
Resetting to NULL might be waranted if we ever thought the class instance might get recreated?